autogluon/utils/tabular/ml/learner/abstract_learner.py

```diff
@@ -172,6 +172,11 @@ def refit_single_full(self, models=None):
         trainer = self.load_trainer()
         return trainer.refit_single_full(X=X, y=y, models=models)
 
+    # Fits _FULL models and links them in the stack so _FULL models only use other _FULL models as input during stacking
```

Comment:
Is the reason for separate `refit_single_full` vs `refit_ensemble_full` methods is because the former just adds one extra new model whereas the latter may modify many models?
Otherwise, I could imagine `refit_single_full` just being a special case of the `refit_ensemble_full` method (ie. only exposing one method to user to apply to both ensembles and bagged base models, and even non-bagged base models).

Reply:
refit_single_full trains a model _FULL without altering its input sources, so a stacker would still use the non-full version models as input. refit_ensemble_full calls refit_single_full, then alters the _FULL models to take other _FULL models as input instead of the normal models. I will try to clarify in documentation.